src/main/java/data/TaskManager.java

```diff
@@ -29,6 +29,14 @@ public void deleteTask(LocalDate date, int taskIndex) {
         }
     }
 
+    public static void updateTask(LocalDate date, int taskIndex, String updatedTask) {
+        List<String> dayTasks = tasks.get(date);
+        if (dayTasks != null && taskIndex >= 0 && taskIndex < dayTasks.size()) {
```

Comment:
Perhaps you can consider extracting these as single booleans, such as dayHasTasks and taskIndexExists.